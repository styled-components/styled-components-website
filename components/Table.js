import styled from 'styled-components'

import rem from '../utils/rem'
import { lightGrey } from '../utils/colors'
import { headerFont } from '../utils/fonts'

export const TableWrapper = styled.table`
  width: 100%;
  text-align: left;
  margin: ${rem(40)} 0;
`

const TableHead = styled.thead`
  border-bottom: 2px solid ${lightGrey};
  font-family: ${headerFont};
`

export const Row = styled.tr`
  padding: 0 ${rem(20)};
  padding-left: 0;

  &:not(:last-child) {
    border-bottom: 1px solid ${lightGrey};
  }
`

export const Column = styled.th`
  font-weight: normal;
  padding: ${rem(10)} ${rem(12)};
  padding-left: 0;
`

const TableHeadColumn = styled(Column)`
  text-transform: uppercase;
  font-size: 85%;
  opacity: 0.8;
`

const Table = ({ head, children }) => (
  <TableWrapper>
    <TableHead>
      <tr>
        {
          head.map((text, i) => (
            <TableHeadColumn key={i} title={text}>
              {text}
            </TableHeadColumn>
          ))
        }
      </tr>
    </TableHead>

    <tbody>
      {children}
    </tbody>
  </TableWrapper>
)

export default Table
