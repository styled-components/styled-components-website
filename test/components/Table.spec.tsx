import React from 'react';
import renderer from 'react-test-renderer';
import Table, { Row, Column } from '../../components/Table';

test('Table renders correctly', () => {
  const tree = renderer
    .create(
      <Table head={['head']}>
        <Row>
          <Column />
        </Row>
      </Table>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
