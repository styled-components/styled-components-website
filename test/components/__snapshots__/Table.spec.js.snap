// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
.c0 {
  width: 100%;
  text-align: left;
  margin: 2.2222222222222223rem 0;
}

.c1 {
  border-bottom: 2px solid rgba(20,20,20,0.1);
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

.c4 {
  padding: 0 1.1111111111111112rem;
  padding-left: 0;
}

.c4:not(:last-child) {
  border-bottom: 1px solid rgba(20,20,20,0.1);
}

.c2 {
  font-weight: normal;
  padding: 0.5555555555555556rem 0.6666666666666666rem;
  padding-left: 0;
}

.c3 {
  text-transform: uppercase;
  font-size: 85%;
  opacity: 0.8;
}

<table
  className="c0"
>
  <thead
    className="c1"
  >
    <tr>
      <th
        className="c2 c3"
        title="head"
      >
        head
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="c4"
    >
      <th
        className="c2"
      />
    </tr>
  </tbody>
</table>
`;
